soci/soci_index_test.go

```diff
@@ -352,3 +353,15 @@ func TestNewIndexFromReader(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildSociLayer(t *testing.T) {
+	ctx := context.Background()
+	desc := ocispec.Descriptor{
+		MediaType: images.MediaTypeDockerSchema2Layer,
+		Digest:    "layerdigest",
+	}
+	_, err := buildSociLayer(ctx, nil, desc, 0, nil, nil)
+	if err.Error() != `layer layerdigest (application/vnd.docker.image.rootfs.diff.tar) must be compressed by gzip, but got "unknown"` {
```

Comment:
This is unnecessarily brittle. Tests should protect functionality, not enshrine current behavior. When we add support for Zstandard layers, this error message will change, and the test--as currently written--will break. (I'll admit that my real complaint here is philosophical, not technical. If developers get in the habit of tweaking unit tests, they learn to ignore test failures even when a failure is trying to tell them about a real problem.)

**TL;DR:** Suggest testing only for the behavior that you want to protect. Something like:
```
        if !strings.Contains(err.Error(), "layerdigest") || !strings.Contains(err.Error(), "application/vnd.docker.image.rootfs.diff.tar") ||
                !strings.Contains(err.Error(), "compress") {
                t.Fatalf("unexpected error: %v", err)

```

Reply:
Sounds good. How about returning ErrFailedPrecondition from https://pkg.go.dev/github.com/containerd/containerd/errdefs in addition to?

Reply:
Or define something like errUnsupportedCompression

Reply:
Would prefer the latter.

Reply:
Something like `errUnsupportedLayerFormat`.